### PR TITLE
Adding Feed Configuration documentation

### PR DESCRIPTION
**Pull request by jeremyf**
## What type of PR is this? (check all applicable)

- [x] Documentation Update

## Description

This documentation outlines the scope and intention of the Feed
Configuration, it's primary purpose is to:

- Reference the historical context
- Establish a draft glossary
- Introduce high-level constraints
- Serve as a document to help with the task break-down

As a note, I'm looking at this document almost through the lens of
[literate programming][1].  In writing the document, I learned about and
attempted to describe the shape of this internal facing project.

Consider for a moment that we are not adding a new user facing feature
to the Forem code base.  Instead we're scoping this project as helping
ease the experimentation process regarding the feed.

In other words the scope of this project is to help name and refine
different feed strategies to understand what each strategy means.

Eventually this could mean that creators and members can have a
future opportunity to select feed strategies that reflect the
aspirational community values of the feed (e.g. we want a feed that
favors comments, or neighborhoods, or the occassional serendipity).

[1]:https://en.wikipedia.org/wiki/Literate_programming

## Related Tickets & Documents

Closes forem/forem#17244
Relates to forem/forem#17245
Informs forem/forem-internal-eng#435

## QA Instructions, Screenshots, Recordings

Read through the document, ask questions, seek clarification.

### UI accessibility concerns?

None.

## Added/updated tests?

- [x] No, and this is why: this is documentation only

## [Forem core team only] How will this change be communicated?

- [x] I will share this change internally with the appropriate teams

### CODE REVIEW

**Review by msarit — APPROVED** [review]

**Review by jgaskins — APPROVED** [review]

File: config/feed/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+## Historical Documents
+
+- The
+  [Articles::Feeds::WeightedQueryStrategy](https://github.com/forem/forem/blob/de2edee07d824a34e5c5445d455b1f8086bd127d/app/services/articles/feeds/weighted_query_strategy.rb)
+  is the precursor to the more robust feed configuration. It provides hints as
+  to how we'll structure the feed configuration.
+- [These Are the [Feed] Levers I Know I Know - Forem Team 🌱](https://forem.team/jeremy/these-are-the-feed-levers-i-know-i-know-293m)
+- [Creating a Catalog of Feed Levers and Configurations - Forem Team 🌱](https://forem.team/jeremy/creating-a-catalog-of-feed-levers-and-configurations-nnk)
+- On <2022-04-11 Mon>, Jeremy, Jennie, Ella, and Michael T met over Zoom to talk
+  through the current feed.
```
Comment (aitchiss) [inline config/feed/README.md:17]:
```suggestion
- On <2022-04-11 Mon>, [Jeremy, Jennie, Ella, and Michael T met over Zoom to talk
  through the current feed](https://forem.team/jeremy/lets-talk-about-the-feed-baby-54b1).
```
Maybe we can link this to your forem.team post with the recording?

File: config/feed/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+## Glossary
+
+- **_Feed Experiment_:** A _feed experiment_ is the current _feed strategies_
+  current configurations we're using to generate feeds
```
Comment (aitchiss) [inline config/feed/README.md:22]:
```suggestion
- **_Feed Experiment_:** A _feed experiment_ is the currently implemented set of _feed strategies_ -
  current configurations we're using to generate feeds
```

When I first read this and the next bullet point I wasn't too clear on the difference between _feed_experiment_ vs _feed_strategies_. I'm not sure if my suggestion captures the distinction we're trying to make here properly, so feel free to ignore if it doesn't seem right 😄

**Review by aitchiss — APPROVED** [review]
Love to see all this great documentation! I've left a few suggestions but feel free to take or leave them 🙂

File: config/feed/README.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+delve into the high-level goals of the configurations, and describe any present
+constraints.
+
+## Historical Documents
+
+- The
+  [Articles::Feeds::WeightedQueryStrategy](https://github.com/forem/forem/blob/de2edee07d824a34e5c5445d455b1f8086bd127d/app/services/articles/feeds/weighted_query_strategy.rb)
+  is the precursor to the more robust feed configuration. It provides hints as
+  to how we'll structure the feed configuration.
+- [These Are the [Feed] Levers I Know I Know - Forem Team 🌱](https://forem.team/jeremy/these-are-the-feed-levers-i-know-i-know-293m)
```
[the remark this answers is not in the document]
Comment (jeremyf) [reply config/feed/README.md:14]: Yes we should!

**Pull request merged**